src/coreclr/jit/flowgraph.cpp

```diff
@@ -2803,6 +2803,64 @@ bool Compiler::fgSimpleLowerCastOfSmpOp(LIR::Range& range, GenTreeCast* cast)
     return false;
 }
 
+//------------------------------------------------------------------------
+// fgSimpleLowerCastOfSmallOp: Optimization to remove CAST nodes from operands of some small ops that depents on
+// lower bits only.
+// Example:
+//      BSWAP16(CAST(x)) transforms to BSWAP16(x)
+//
+// Returns:
+//      True or false, representing changes were made.
+//
+// Notes:
+//      This optimization could be done in morph, but it cannot because there are correctness
+//      problems with NOLs (normalized-on-load locals) and how they are handled in VN.
+//      Simple put, you cannot remove a CAST from CAST(LCL_VAR{nol}) in HIR.
+//
+//      Because the optimization happens during rationalization, turning into LIR, it is safe to remove the CAST.
+//
+bool Compiler::fgSimpleLowerCastOfSmallOp(LIR::Range& range, GenTree* op)
```

Comment:
I would just call this `fgSimpleLowerBswap16` when it is only optimizing `BSWAP16`.